### DISCUSSION TITLE
DW.Android.Helpers Crash

### DISCUSSION THREAD

**Discussion by radlik01** [DelphiWorlds/Kastri#176] [General]
Dear all,

I have faced another problem with Android service. When I use the unit DW.Android.Helpers my android service crash on startup.
I use Delphi 11.3 and Android 8.1 device.

in the adb log is this dump

`06-07 09:44:07.000 28128 28128 I crash_dump64: performing dump of process 28098 (target tid = 28098)
06-07 09:44:07.000 28098 28098 I libc    : orig_dumpable: 1
06-07 09:44:07.001 28128 28128 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
06-07 09:44:07.001 28128 28128 F DEBUG   : Build fingerprint: 'HUAWEI/DRA-L21/HWDRA-M:8.1.0/HUAWEIDRA-L21/1.0.0.140(C432):user/release-keys'
06-07 09:44:07.001 28128 28128 F DEBUG   : Revision: '0'
06-07 09:44:07.001 28128 28128 F DEBUG   : ABI: 'arm64'
06-07 09:44:07.001 28128 28128 F DEBUG   : pid: 28098, tid: 28098, name: .SIDSafeTracker  >>> eu.fss.SIDSafeTracker <<<
06-07 09:44:07.001 28128 28128 F DEBUG   : signal 11 (SIGSEGV), code 2 (SEGV_ACCERR), fault addr 0x73357dbb68
06-07 09:44:07.001 28128 28128 F DEBUG   :     x0   910003fda9bf7bfd  x1   002f00640069006f  x2   0000000000000000  x3   000000734a2de800
06-07 09:44:07.001 28128 28128 F DEBUG   :     x4   0000000000000000  x5   000000734a33abfa  x6   0072006500530054  x7   0065006300690076
06-07 09:44:07.001 28128 28128 F DEBUG   :     x8   0000000000000000  x9   00000073357dbb68  x10  fffffffffffffff0  x11  0000000000000018
06-07 09:44:07.002 28128 28128 F DEBUG   :     x12  000000733467e8d0  x13  0000000000000003  x14  00000000ffffffff  x15  aaaaaaaaaaaaaaab
06-07 09:44:07.002 28128 28128 F DEBUG   :     x16  00000073d85ea2c0  x17  00000073d858aba0  x18  00000073d85f5000  x19  0000007356c5fa00
06-07 09:44:07.002 28128 28128 F DEBUG   :     x20  0000007356acc990  x21  0000007356c5fa00  x22  0000007fc1b25eac  x23  00000073383119d1
06-07 09:44:07.002 28128 28128 F DEBUG   :     x24  0000000000000008  x25  00000073dc231a40  x26  0000007356c5faa0  x27  0000000000000002
06-07 09:44:07.002 28128 28128 F DEBUG   :     x28  0000000000000000  x29  0000007fc1b256e0  x30  00000073357e01b4
06-07 09:44:07.002 28128 28128 F DEBUG   :     sp   0000007fc1b256b0  pc   0000007335702638  pstate 00000000a0000000
06-07 09:44:07.010 28128 28128 F DEBUG   :
06-07 09:44:07.010 28128 28128 F DEBUG   : backtrace:
06-07 09:44:07.011 28128 28128 F DEBUG   :     #00 pc 0000000000b7f638  /data/app/eu.fss.SIDSafeTracker-5G6UitfOLSRkjwT5BAR4Kg==/lib/arm64/libSIDSafeTrackerService.so
06-07 09:44:07.011 28128 28128 F DEBUG   :     #01 pc 0000000000c5d1b0  /data/app/eu.fss.SIDSafeTracker-5G6UitfOLSRkjwT5BAR4Kg==/lib/arm64/libSIDSafeTrackerService.so
06-07 09:44:07.011 28128 28128 F DEBUG   :     #02 pc 0000000000c58ba0  /data/app/eu.fss.SIDSafeTracker-5G6UitfOLSRkjwT5BAR4Kg==/lib/arm64/libSIDSafeTrackerService.so
06-07 09:44:07.011 28128 28128 F DEBUG   :     #03 pc 0000000000d467ec  /data/app/eu.fss.SIDSafeTracker-5G6UitfOLSRkjwT5BAR4Kg==/lib/arm64/libSIDSafeTrackerService.so
06-07 09:44:07.011 28128 28128 F DEBUG   :     #04 pc 0000000000d2ff0c  /data/app/eu.fss.SIDSafeTracker-5G6UitfOLSRkjwT5BAR4Kg==/lib/arm64/libSIDSafeTrackerService.so
06-07 09:44:07.011 28128 28128 F DEBUG   :     #05 pc 0000000000d37a2c  /data/app/eu.fss.SIDSafeTracker-5G6UitfOLSRkjwT5BAR4Kg==/lib/arm64/libSIDSafeTrackerService.so
06-07 09:44:07.011 28128 28128 F DEBUG   :     #06 pc 0000000000d3feb0  /data/app/eu.fss.SIDSafeTracker-5G6UitfOLSRkjwT5BAR4Kg==/lib/arm64/libSIDSafeTrackerService.so
06-07 09:44:07.011 28128 28128 F DEBUG   :     #07 pc 0000000000d404cc  /data/app/eu.fss.SIDSafeTracker-5G6UitfOLSRkjwT5BAR4Kg==/lib/arm64/libSIDSafeTrackerService.so
06-07 09:44:07.011 28128 28128 F DEBUG   :     #08 pc 0000000000e8e07c  /data/app/eu.fss.SIDSafeTracker-5G6UitfOLSRkjwT5BAR4Kg==/lib/arm64/libSIDSafeTrackerService.so
06-07 09:44:07.011 28128 28128 F DEBUG   :     #09 pc 0000000000f1ed98  /data/app/eu.fss.SIDSafeTracker-5G6UitfOLSRkjwT5BAR4Kg==/lib/arm64/libSIDSafeTrackerService.so (OnCreateDelphi+260)
06-07 09:44:07.011 28128 28128 F DEBUG   :     #10 pc 00000000002ae270  /data/app/eu.fss.SIDSafeTracker-5G6UitfOLSRkjwT5BAR4Kg==/lib/arm64/libProxyAndroidService.so (Java_com_embarcadero_rtl_ProxyService_onCreateNative+164)
06-07 09:44:07.011 28128 28128 F DEBUG   :     #11 pc 0000000000035748  /data/app/eu.fss.SIDSafeTracker-5G6UitfOLSRkjwT5BAR4Kg==/oat/arm64/base.odex (offset 0x35000)`

Any idea ?

Thank you

Jan

**Comment by DelphiWorlds** [comment]
Thank you so much for giving me the clue I needed to solve this issue: https://github.com/DelphiWorlds/Kastri/issues/175

I've now pushed a fix for it (changed `DW.Toast.Android`)

**Reply by radlik01** [reply]
Thank you. Now it works
